# Python Lab #3 : Pandas

# EXERCISE #1

## Step 1.1 : Load the data file into a data frame, setting the Rank column as the index

In [1]:
import numpy as np
import pandas as pd
from pandas import Series, DataFrame
vgsales = pd.read_csv('vgsales.csv', index_col = 'Rank')

## Step 1.2 : Preview the first ten titles in the data set

In [2]:
vgsales['Name'].head(10)

Rank
1                    Wii Sports
2             Super Mario Bros.
3                Mario Kart Wii
4             Wii Sports Resort
5      Pokemon Red/Pokemon Blue
6                        Tetris
7         New Super Mario Bros.
8                      Wii Play
9     New Super Mario Bros. Wii
10                    Duck Hunt
Name: Name, dtype: object

## Step 1.3 : Report the total number of titles

In [3]:
print("The total number of titles = {}" .format(vgsales['Name'].count()))

The total number of titles = 16598


## Step 1.4 : Report some summary statistics about the distribution of global sales

In [4]:
print("The mean is : {}" .format(np.mean(vgsales['Global_Sales'])))
print("The median is : {}" .format(np.median(vgsales['Global_Sales'])))
print("The standard deviation is : {}" .format(np.std(vgsales['Global_Sales'])))

The mean is : 0.53744065550074
The median is : 0.17
The standard deviation is : 1.5549810910296504


## Step 1.5 : What proportion (or percentage) of titles earned at least 1 million dollars in global sales?

In [5]:
print("The  proportion of titles that earned at least $1 million in global sales : {}" .format((vgsales['Global_Sales'][vgsales['Global_Sales'] >= 1].count())/vgsales['Name'].count()))

The  proportion of titles that earned at least $1 million in global sales : 0.12537655139173395


## The proportion of titles that earned at least 1 million in global sales is 0.12538%
## The percentage is of titles that earned at least 1 million in global sales 12.538%

# EXERCISE #2

## Step 2.1 : Filter the DataFrame to show  the top 3 sellers in the Action genre

In [6]:
print ('The top 3 sellers in the Action genre are:')
print (vgsales[vgsales['Genre'] == 'Action'].sort_values(by = 'Global_Sales',ascending = False)['Name'].head(n=3))

The top 3 sellers in the Action genre are:
Rank
17               Grand Theft Auto V
18    Grand Theft Auto: San Andreas
24               Grand Theft Auto V
Name: Name, dtype: object


## Step 2.2 : Generalize the filter you created above and write a function top_sellers for the same

In [7]:
def top_sellers (vgsales,cat=None,val=None,n=5):
    if cat != None and val != None:
        df_top = vgsales[vgsales[cat] == val].head(n)
    else:
        df_top = vgsales.head(n)
    return df_top

## Step 2.3: Use the function to display the overall top 5 sellers

In [8]:
top_sellers(vgsales)

Name Platform    Year         Genre Publisher  \
Rank                                                                      
1                   Wii Sports      Wii  2006.0        Sports  Nintendo   
2            Super Mario Bros.      NES  1985.0      Platform  Nintendo   
3               Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
4            Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
5     Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   

      NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
Rank                                                           
1        41.49     29.02      3.77         8.46         82.74  
2        29.08      3.58      6.81         0.77         40.24  
3        15.85     12.88      3.79         3.31         35.82  
4        15.75     11.01      3.28         2.96         33.00  
5        11.27      8.89     10.22         1.00         31.37

## Use the function to display the top 10 NES games

In [9]:
top_sellers(vgsales,'Platform','NES',10)

Name Platform    Year      Genre Publisher  \
Rank                                                                          
2                   Super Mario Bros.      NES  1985.0   Platform  Nintendo   
10                          Duck Hunt      NES  1984.0    Shooter  Nintendo   
23                Super Mario Bros. 3      NES  1988.0   Platform  Nintendo   
97                Super Mario Bros. 2      NES  1988.0   Platform  Nintendo   
128               The Legend of Zelda      NES  1986.0     Action  Nintendo   
156                            Tetris      NES  1988.0     Puzzle  Nintendo   
216                         Dr. Mario      NES  1990.0     Puzzle  Nintendo   
252   Zelda II: The Adventure of Link      NES  1987.0  Adventure  Nintendo   
277      Teenage Mutant Ninja Turtles      NES  1989.0     Action    Palcom   
278                        Excitebike      NES  1984.0     Racing  Nintendo   

      NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
Rank                                                           
2        29.08      3.58      6.81         0.77         40.24  
10       26.93      0.63      0.28         0.47         28.31  
23        9.54      3.44      3.84         0.46         17.28  
97        5.39      1.18      0.70         0.19          7.46  
128       3.74      0.93      1.69         0.14          6.51  
156       2.97      0.69      1.81         0.11          5.58  
216       2.62      0.60      1.52         0.10          4.85  
252       2.19      0.50      1.61         0.08          4.38  
277       3.38      0.44      0.31         0.04          4.17  
278       2.04      0.48      1.57         0.07          4.16

## Use the function to display top 10 selling games in the year 2000

In [10]:
top_sellers(vgsales,'Year',2000,10)

Name Platform    Year         Genre  \
Rank                                                                      
133              Pokémon Crystal Version       GB  2000.0  Role-Playing   
174                     Final Fantasy IX       PS  2000.0  Role-Playing   
224                             Driver 2       PS  2000.0        Action   
226             Tony Hawk's Pro Skater 2       PS  2000.0        Sports   
243   Dragon Quest VII: Warriors of Eden       PS  2000.0  Role-Playing   
295                Tekken Tag Tournament      PS2  2000.0      Fighting   
334            Spyro: Year of the Dragon       PS  2000.0      Platform   
359                       WWF SmackDown!       PS  2000.0      Fighting   
368          Rugrats in Paris: The Movie       PS  2000.0        Action   
394                           Crash Bash       PS  2000.0          Misc   

                        Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  \
Rank                                                                           
133                      Nintendo      2.55      1.56      1.29         0.99   
174                    SquareSoft      1.62      0.77      2.78         0.14   
224                         Atari      2.36      2.10      0.02         0.25   
226                    Activision      3.05      1.41      0.02         0.20   
243              Enix Corporation      0.20      0.14      4.10         0.02   
295            Namco Bandai Games      1.68      1.51      0.51         0.35   
334   Sony Computer Entertainment      1.93      1.58      0.00         0.19   
359                           THQ      2.01      1.35      0.06         0.16   
368                           THQ      1.96      1.33      0.00         0.23   
394   Sony Computer Entertainment      1.56      1.47      0.19         0.17   

      Global_Sales  
Rank                
133           6.39  
174           5.30  
224           4.73  
226           4.68  
243           4.47  
295           4.05  
334           3.71  
359           3.58  
368           3.52  
394           3.39

## Use the function to determine top 10 selling sports games

In [11]:
top_sellers(vgsales,'Genre','Sports',10)

Name Platform    Year   Genre  \
Rank                                                                
1                             Wii Sports      Wii  2006.0  Sports   
4                      Wii Sports Resort      Wii  2009.0  Sports   
14                               Wii Fit      Wii  2007.0  Sports   
15                          Wii Fit Plus      Wii  2009.0  Sports   
78                               FIFA 16      PS4  2015.0  Sports   
86    Mario & Sonic at the Olympic Games      Wii  2007.0  Sports   
113                              FIFA 14      PS3  2013.0  Sports   
118                        Zumba Fitness      Wii  2010.0  Sports   
122                              FIFA 12      PS3  2011.0  Sports   
125                              FIFA 15      PS4  2014.0  Sports   

            Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
Rank                                                                            
1            Nintendo     41.49     29.02      3.77         8.46         82.74  
4            Nintendo     15.75     11.01      3.28         2.96         33.00  
14           Nintendo      8.94      8.03      3.60         2.15         22.72  
15           Nintendo      9.09      8.59      2.53         1.79         22.00  
78    Electronic Arts      1.11      6.06      0.06         1.26          8.49  
86               Sega      2.58      3.90      0.66         0.91          8.06  
113   Electronic Arts      0.78      4.32      0.07         1.73          6.90  
118         505 Games      3.50      2.64      0.00         0.67          6.81  
122   Electronic Arts      0.84      4.32      0.11         1.42          6.69  
125   Electronic Arts      0.79      4.29      0.05         1.47          6.59

# EXERCISE #3

## Step 3.1 : Generate a (linear) correlation matrix between sales in specific markets (North America, Europe, Japan, Other) and global sales.

In [12]:
sales = vgsales[vgsales.columns[-6:]]
sales.corr()

NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales
NA_Sales      1.000000  0.767727  0.449787     0.634737      0.941047
EU_Sales      0.767727  1.000000  0.435584     0.726385      0.902836
JP_Sales      0.449787  0.435584  1.000000     0.290186      0.611816
Other_Sales   0.634737  0.726385  0.290186     1.000000      0.748331
Global_Sales  0.941047  0.902836  0.611816     0.748331      1.000000

## Relation of sales between various markets is positively correlated.

## Step 3.2 : Repeat this calculation for the subset of video games released in the year 2000, and compare these results to those for the subset of video games released in the year 2015. What do you observe about how the relationship between the sales figures has changed over the 15-year period?

In [13]:
data = pd.read_csv("vgsales.csv")
year_2000=data.loc[data['Year'] == 2000.0]
year_2000_sales = year_2000[year_2000.columns[-6:]]
print("Correlation matrix in the year 2000 : ")
print()
print(year_2000_sales.corr())
print()
print("Correlation matrix in the year 2015 : ")
print()
year_2015=data.loc[data['Year'] == 2015.0]
year_2015_sales = year_2015[year_2015.columns[-6:]]
print(year_2015_sales.corr())

Correlation matrix in the year 2000 : 

              NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales
NA_Sales      1.000000  0.866555  0.139115     0.691317      0.900815
EU_Sales      0.866555  1.000000  0.108401     0.748552      0.866616
JP_Sales      0.139115  0.108401  1.000000     0.226852      0.512446
Other_Sales   0.691317  0.748552  0.226852     1.000000      0.761891
Global_Sales  0.900815  0.866616  0.512446     0.761891      1.000000

Correlation matrix in the year 2015 : 

              NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales
NA_Sales      1.000000  0.736844  0.123217     0.866672      0.911555
EU_Sales      0.736844  1.000000  0.132160     0.926506      0.922387
JP_Sales      0.123217  0.132160  1.000000     0.126803      0.306981
Other_Sales   0.866672  0.926506  0.126803     1.000000      0.956540
Global_Sales  0.911555  0.922387  0.306981     0.956540      1.000000


## The global sales correlation coefficient for the year 2015 in NA was higher than it was in 2000
## The global sales correlation coefficient for the year 2015 in EU was higher than it was in 2000
## The global sales correlation coefficient for the year 2015 in JP was lower than it was in 2000
## The global sales correlation coefficient for the year 2015 in Other regions was higher in than it was in 2000